pkg/cache/json.go

```diff
@@ -165,13 +165,14 @@ func (q *JSON) existingDigest() (string, error) {
 }
 
 func (q *JSON) computeDigest(fbcFsys fs.FS) (string, error) {
+	buf := make([]byte, 32*1024)
```

Comment:
I left a comment in slack too, but my only nit on this line (and the same kind of thing in tar.go is that I'm super not-a-fan of magic numbers here.  I buy your argument, so this isn't a sizing/allocation issue.